docs/_integration-with-existing-apps-ios.md

```diff
@@ -0,0 +1,598 @@
+import UIKit
++import React_RCTAppDelegate
++import React_RCTAppDelegate
+
+@main
+-class AppDelegate: UIResponder, UIApplicationDelegate {
++class AppDelegate: RCTAppDelegate {
+
+-  var window: UIWindow?
+
+-  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
++  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    // Override point for customization after application launch.
++    self.automaticallyLoadReactNativeWindow = false
++    super.application(application, didFinishLaunchingWithOptions: launchOptions)
+    window = UIWindow()
+-    window?.rootViewController = ViewController()
+-    window?.makeKeyAndVisible()
++    window.rootViewController = ViewController()
++    window.makeKeyAndVisible()
+    return true
+  }
+
++  override func sourceURL(for bridge: RCTBridge) -> URL? {
++    self.bundleURL()
++  }
+
++  override func bundleURL() -> URL? {
++#if DEBUG
++    RCTBundleURLProvider.sharedSettings().jsBundleURL(forBundleRoot: "index")
++#else
++    Bundle.main.url(forResource: "main", withExtension: "jsbundle")
++#endif
++  }
+}
+```
+
+Let's have a look at the code above:
+
+1. We are inherithing from the `RCTAppDelegate` and we are calling the `applicatinon(_:didFinishLaunchingWithOptions:)` of the `RCTAppDelegate`. This delegates all the React Native initialization processes to the base class.
+2. We are customizing the `RCTAppDelegate` by setting the `automaticallyLoadReactNativeWindow` to `false`. This step instruct React Native that the app is handling the `UIWindow` and React Native should not worry about that.
+3. The methods `sourceURLForBridge(for:)` and `bundleURL()` are used by the App to tell to React Native wher it can find the JS bundle that needs to be rendered. The `sourceURLForBridge(for:)` is from the Old Architecture and you can see that it is deferring the decision to the `bundleURL()` method, required by the New Architecture.
+
+</TabItem>
+</Tabs>
+
+#### Presenting a React Native view in a rootViewController
+
+Finally, we can present our React Native view. To do so, we need a new View Controller that can host a view in which we can load the JS content.
+
+1. From Xcode, let's create a new `UIViewController` (Let's call it `ReactViewController`).
+2. Have the Initial `ViewController` present the `ReactViewController`. There are several ways to do so, depending on your app. For this example, we assume that you have a button that presents React Native Modally.
+
+<Tabs groupId="ios-language" queryString defaultValue={constants.defaultAppleLanguage} values={constants.appleLanguages}>
+<TabItem value="objc">
+
+```diff title="ViewController.m"
+#import "ViewController.h"
++#import "ReactViewController.h"
+
+@interface ViewController ()
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
```

Comment:
Is this correct with the `-`?

Reply:
yes, in objc this denotes an instance method. a `+` instead, denotes a static method. Yes, the language is weird.